egg/core/language_analysis.py

```diff
@@ -124,3 +149,70 @@ def compute_distance(_list, distance):
 
         output_message = json.dumps(dict(topsim=topsim, epoch=epoch))
         print(output_message, flush=True)
+
+
+class PosDisent(Callback):
+    """
+    Positional disentanglement metric, introduced in "Compositionality and Generalization in Emergent Languages", 
+    Chaabouni et al., ACL 2020.
+    """
+    def __init__(self, print_train: bool = True, is_gumbel: bool = False):
+        super().__init__()
+        self.print_train = print_train
```

Comment:
I've found computing posdis on the train set to slow down training quite a lot for the multitask multiagent game. In genera, wouldn't it be better to have it computed on the validation data only by default?

Reply:
Done